src/Std/Data/DHashMap/Lemmas.lean

```diff
@@ -1512,6 +1512,94 @@ end Const
 
 end monadic
 
+theorem any_eq_toList_any {p : (a : α) → β a → Bool} :
+    m.any p = m.toList.any (fun x => p x.1 x.2) := Raw₀.any_eq_toList_any ⟨m.1, m.2.size_buckets_pos⟩
+
+theorem any_eq_true_iff_exists_mem_get [LawfulBEq α] {p : (a : α) → β a → Bool} :
+    m.any p = true ↔ ∃ (a : α) (h : a ∈ m), p a (m.get a h) :=
+  Raw₀.any_eq_true ⟨m.1, m.2.size_buckets_pos⟩ m.2
+
+theorem any_eq_false_iff_forall_mem_get [LawfulBEq α] {p : (a : α) → β a → Bool} :
+    m.any p = false ↔ ∀ (a : α) (h : a ∈ m), p a (m.get a h) = false :=
+  Raw₀.any_eq_false ⟨m.1, m.2.size_buckets_pos⟩ m.2
+
+theorem all_eq_toList_all {p : (a : α) → β a → Bool} :
+    m.all p = m.toList.all (fun x => p x.1 x.2) := Raw₀.all_eq_toList_all ⟨m.1, m.2.size_buckets_pos⟩
+
+theorem all_eq_neg_any_neg {p : (a : α) → β a → Bool} :
+    m.all p = ! m.any (fun a b => ! p a b) := Raw₀.all_eq_neg_any_neg ⟨m.1, m.2.size_buckets_pos⟩
+
+theorem any_eq_neg_all_neg {p : (a : α) → β a → Bool} :
+    m.any p = ! m.all (fun a b => ! p a b) := Raw₀.any_eq_neg_all_neg ⟨m.1, m.2.size_buckets_pos⟩
```

Comment:
The recommended spelling of `!` is `not`.

(I would also have suggested to state these lemmas the other way around, but `Array.any_eq_not_all_not` also exists, so I agree that this is the more consistent choice.)